### PR TITLE
RTEMIS-182: Student mapping with School & Lottery Status Table.

### PR DESCRIPTION
**Pull request by RohanSingh-inno**
Ticket # : https://innoraft.atlassian.net/browse/RTEMIS-182

---
### What the PR does
- Created a custom lottery status table.
- Created config from for lottery status table data deletion.
- Installed Ultimate cron module.
- Added a student reference field to the school mini node.

### CODE REVIEW

**Review by kunal-sin — APPROVED** [review]

File: modules/rte_mis_lottery/config/install/rte_mis_lottery.settings.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+time_interval: ''
```
Comment (jaswinsingh) [inline modules/rte_mis_lottery/config/install/rte_mis_lottery.settings.yml:1]: Please add default time in config.

File: modules/rte_mis_lottery/rte_mis_lottery.install [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        'default' => 0,
+      ],
+      'mobile_number' => [
+        'description' => 'Student mobile number',
+        'type' => 'varchar',
+        'length' => 20,
+        'not null' => TRUE,
+      ],
+      'alloted_school' => [
+        'description' => 'Student Alotted school',
```
Comment (jaswinsingh) [inline modules/rte_mis_lottery/rte_mis_lottery.install:35]: Allotted*

File: modules/rte_mis_lottery/rte_mis_lottery.install [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      ],
+      'student_application_number' => [
+        'description' => 'Student Application Number',
+        'type' => 'int',
+        'unsigned' => TRUE,
+        'not null' => TRUE,
+        'default' => 0,
+      ],
+      'mobile_number' => [
+        'description' => 'Student mobile number',
```
Comment (jaswinsingh) [inline modules/rte_mis_lottery/rte_mis_lottery.install:29]: Change the label to mobile number.

File: modules/rte_mis_lottery/rte_mis_lottery.install [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+ * Install, update and uninstall functions for this module.
+ */
+
+/**
+ * Implements hook_schema().
+ */
+function rte_mis_lottery_schema() {
+  $schema['lottery_status'] = [
+    'description' => 'Stores the status of the lottery results',
+    'fields' => [
```
[the remark this answers is not in the document]
Comment (RohanSingh-inno) [reply modules/rte_mis_lottery/rte_mis_lottery.install:14]: Ok

File: modules/rte_mis_lottery/config/install/rte_mis_lottery.settings.yml (hunk 1 above)
Comment (RohanSingh-inno) [reply modules/rte_mis_lottery/config/install/rte_mis_lottery.settings.yml]: Added 24 hours as default

File: modules/rte_mis_lottery/rte_mis_lottery.install (hunk 2 above)
Comment (RohanSingh-inno) [reply modules/rte_mis_lottery/rte_mis_lottery.install]: ok

File: modules/rte_mis_lottery/rte_mis_lottery.install (hunk 3 above)
Comment (RohanSingh-inno) [reply modules/rte_mis_lottery/rte_mis_lottery.install]: Ok

File: modules/rte_mis_lottery/src/Services/ClearTableManager.php [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+   */
+  public function __construct(Connection $database) {
+    $this->database = $database;
+  }
+
+  /**
+   * Truncates all the entries from the custom `lottery_status` table.
+   */
+  public function clearTable() {
+    $this->database->truncate('lottery_status')->execute();
```
[the remark this answers is not in the document]
Comment (RohanSingh-inno) [reply modules/rte_mis_lottery/src/Services/ClearTableManager.php]: Ok

File: modules/rte_mis_lottery/src/Form/LotteryConfigForm.php [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+    $form['time_interval'] = [
+      '#type' => 'number',
+      '#title' => $this->t('Time Interval'),
+      '#description' => $this->t('Enter the time interval in hours after which the lottery status table will be cleared.'),
+      '#default_value' => $config->get('time_interval'),
+      '#min' => 1,
+      '#required' => TRUE,
+    ];
+
```
Comment (jaswinsingh) [inline modules/rte_mis_lottery/src/Form/LotteryConfigForm.php:87]: Is this accepting float value? Can we add validation of minimum value of 24 hours?

**Review by jaswinsingh — APPROVED** [review]

**Pull request merged**